### PR TITLE
Eos 21918: Logging for Node health collection

### PR DESCRIPTION
**Pull request by gargpalak048**
Logging for Node health collection

Add Logging module for Node-Health Feature

## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [ ] Did your address all Codacy issues?

## Testing:

* [ ] Unittest updated and all unittests are passing?

* [ ] Sanity/Self tests are updated (if applicable)?

* [ ] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py [hunk 1]
```diff
@@ -29,7 +32,9 @@ class ResourceMap(metaclass=ABCMeta):
 
     def __init__(self):
         """Initialize resource."""
-        pass
+        logging_level = Conf.get(SSPL_CONF,
+            f"{SYSTEM_INFORMATION}>{LOG_LEVEL}", "INFO")
+        init_logging('sspl_node-health', logging_level)
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py:37]: just keep 'node_health'

File: low-level/framework/utils/utility.py [hunk 2]
```diff
@@ -167,3 +167,11 @@ def errno_to_str_mapping(err_no):
         logger.error("map_errno_to_text errno to text mapping failed due to %s: %s " % err)
         return
 
+class CustomLog:
```
Comment (cdeshmukh) [inline low-level/framework/utils/utility.py:170]: Cant we keep in existing logger class

File: unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_server.py [hunk 3]
```diff
@@ -58,14 +58,20 @@ def get_sdr_type_response(cmd):
...
+
+    @classmethod
+    def create_server_obj(cls):
+        if cls._server_map is None:
+            cls._server_map = ServerMap()
+        return cls._server_map
 
     @patch("framework.utils.ipmi_client.IPMITool._run_ipmitool_subcommand")
     def test_build_platform_sensor_data_temperature(self, sdr_type):
+        obj = self.create_server_obj()
```
Comment (cdeshmukh) [inline unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_server.py:72]: keep useful obj name, like previous

File: unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_storage.py [hunk 4]
```diff
@@ -11,15 +11,21 @@
...
+    def create_storage_obj(cls):
+        if cls._storage_map is None:
+            cls._storage_map = StorageMap()
+        return cls._storage_map
 
     @patch(
         "files.opt.seagate.sspl.setup.resource_map.lr2.storage.StorageMap.get_realstor_encl_data"
     )
     def test_get_platform_sensors(self, encl_response):
+        obj = self.create_storage_obj()
```
Comment (cdeshmukh) [inline unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_storage.py:26]: same here

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py (hunk 1 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py]: done

File: low-level/framework/utils/utility.py (hunk 2 above)
Comment (gargpalak048) [reply low-level/framework/utils/utility.py]: done

File: unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_server.py (hunk 3 above)
Comment (gargpalak048) [reply unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_server.py]: done

File: unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_storage.py (hunk 4 above)
Comment (gargpalak048) [reply unittests/files/opt/seagate/sspl/setup/resource_map/lr2/test_storage.py]: done

File: low-level/framework/utils/utility.py [hunk 5]
```diff
@@ -166,4 +166,3 @@ def errno_to_str_mapping(err_no):
     except Exception as err:
         logger.error("map_errno_to_text errno to text mapping failed due to %s: %s " % err)
         return
-
```
Comment (cdeshmukh) [inline low-level/framework/utils/utility.py:169]: nothing changed here, this file should not be there in this PR

**Review by cdeshmukh — COMMENTED** [review]
resolve conflicts plz

File: low-level/framework/utils/utility.py (hunk 5 above)
Comment (gargpalak048) [reply low-level/framework/utils/utility.py:169]: Yes.. no change in this file. even #L169 is also there at my dev machine.
Still it is showing here

**Comment by gargpalak048** [comment]
> resolve conflicts plz

Done. Rebased the branch with latest changes

**Review by cdeshmukh — APPROVED** [review]

**Review by sandeepsgt — APPROVED** [review]

**Review by shriya-deshmukh — APPROVED** [review]

**Review by satish-seagate — APPROVED** [review]
@gargpalak048, Verify logging is properly working or not when the SSPL service is stoped.

**Comment by gargpalak048** [comment]
> @gargpalak048, Verify logging is properly working or not when the SSPL service is stoped.

@satish-seagate verified, logs gets generated properly when sspl is not running

**Review by satish-seagate — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**